### PR TITLE
adding arrow navigation to search

### PR DESCRIPTION
**Pull request by dniziolek**
screenshot from 2016-12-16 17-18-05

### CODE REVIEW

File: src/docs/_includes/navigation.html [hunk 1]
```diff
@@ -58,7 +69,7 @@
     });
   }
 
-  $searchBox.on('input', function() {
+  $searchBox.on('input', function(e) {
```
Comment (kdzwinel) [inline src/docs/_includes/navigation.html]: if `e` isn't used, please remove it

File: src/docs/_includes/navigation.html [hunk 2]
```diff
@@ -45,6 +46,16 @@
   var $searchBox = $('#js-search-box');
   var $searchResults = $('#js-search-results');
 
+  function removeClassActive(index) {
+    var searchResults = $searchResults[0];
```
Comment (kdzwinel) [inline src/docs/_includes/navigation.html]: IMO you should move it under `var $searchResults` (three lines up)

File: src/docs/_includes/navigation.html [hunk 3]
```diff
@@ -77,4 +88,61 @@
...
+          addClassActive(active);
+        }
+      }
+
+      if (e.keyCode === 38) {   //arrowup
+        e.preventDefault();
+        previousActive = active;
+        active--;
+
+        console.log(active);
```
Comment (kdzwinel) [inline src/docs/_includes/navigation.html]: ditto

File: src/docs/_includes/navigation.html [hunk 4]
```diff
@@ -77,4 +88,61 @@
...
+          removeClassActive(previousActive);
+        } else {
+          removeClassActive(previousActive);
+          addClassActive(active);
+        }
+      }
+
+      if (e.keyCode === 13) {   //enter
+        var searchResults = $searchResults[0];
+        var href = searchResults.children[active].querySelector('a').getAttribute("href");
```
Comment (kdzwinel) [inline src/docs/_includes/navigation.html]: please use `'` instead of `"` where possible

File: src/docs/_includes/navigation.html [hunk 5]
```diff
@@ -77,4 +87,58 @@
     });
   });
 
+  $searchBox.on('keydown', function(e) {
+      var children = $searchResults.children();
+
+      if (e.keyCode === 40) {  //arrowdown
```
Comment (matzimowski) [inline src/docs/_includes/navigation.html]: what about keeping `keyCode`s as `const`s at the very beginning (no additional comments needed then)?

File: src/docs/_includes/navigation.html [hunk 6]
```diff
@@ -44,6 +45,15 @@
   // search
   var $searchBox = $('#js-search-box');
   var $searchResults = $('#js-search-results');
+  var searchResults = $searchResults[0];
+
+  function removeClassActive(index) {
+    searchResults.children[index].classList.remove('active');
```
Comment (matzimowski) [inline src/docs/_includes/navigation.html]: what about checking element exists or at least if `index > 0`?

File: src/docs/_sass/_navigation.scss [hunk 7]
```diff
@@ -58,6 +58,10 @@
             background: $graySecondary;
           }
         }
+
+        &.active {
```
Comment (vergilius) [inline src/docs/_sass/_navigation.scss]: change to modifier class (`&--active`)

File: src/docs/_includes/navigation.html [hunk 8]
```diff
@@ -77,4 +87,58 @@
...
 
+  $searchBox.on('keydown', function(e) {
+      var children = $searchResults.children();
+
+      if (e.keyCode === 40) {  //arrowdown
+        e.preventDefault();
+        var previousActive = active;
+        active++;
+
+        if (active == children.length) {
```
Comment (vergilius) [inline src/docs/_includes/navigation.html]: `===`

File: src/docs/_includes/navigation.html [hunk 9]
```diff
@@ -48,11 +48,19 @@
   var searchResults = $searchResults[0];
 
   function removeClassActive(index) {
-    searchResults.children[index].classList.remove('active');
+    if (index >= 0) {
+      searchResults.children[index].classList.remove('active');
+    } else {
+      return;
+    }
```
Comment (kdzwinel) [inline src/docs/_includes/navigation.html]: this `else { return; }` does nothing. Function will return anyway.

File: src/docs/_includes/navigation.html [hunk 10]
```diff
@@ -48,11 +48,19 @@
...
   }
 
   function addClassActive(index) {
-    searchResults.children[index].classList.add('active');
+    if (index >= 0) {
+      searchResults.children[index].classList.add('active');
+    } else {
+      return;
+    }
```
Comment (kdzwinel) [inline src/docs/_includes/navigation.html]: ditto

**Pull request merged**